### DISCUSSION TITLE
cannot reffer to dict value when I use use "is_on=self.bind()" on Switch feature

### DISCUSSION THREAD

**Discussion by Nachsh** [rio-labs/rio#179] [Q&A] [answered]
I wish to use a dict in order to hold and track my switches state. for doing so I create a dict of binaries and I want to use in the "on_change". 

I receive the following error during build:
   " is_on=self.bind().led_status['red'],
TypeError: 'PleaseTurnThisIntoAnAttributeBinding' object is not subscriptable "

code:
"
led_status: dict = {'red':False,'green':False,'blue':False,'ir':False}

 rio.Switch(
                is_on=self.bind().led_status['red'],
                on_change=self.on_led_change,
            ),        
"

**Comment by Aran-Fey** [comment, marked as answer]
Attribute bindings cannot be combined with any extra code. In other words, `is_on=self.bind().led_status` is valid, but `is_on=self.bind().led_status['red']` is not. You will need to update your dict in your `on_change` event handler, like so:

```python

class Example(rio.Component):
    led_status: dict = {
        "red": False,
        "green": False,
        "blue": False,
        "ir": False,
    }

    def on_led_change(self, event: rio.SwitchChangeEvent):
        self.led_status['red'] = event.is_on

    def build(self):
        return rio.Switch(
            is_on=self.led_status["red"],
            on_change=self.on_led_change,
        )
```

**Reply by Nachsh** [reply]
Thanks @Aran-Fey , I was hoping to avoid this implementation as it force me to build multiple "on_led_change" functions as I have multiple switches. But, I'll manage if no other alternative exist.
